interoperator/pkg/controller/multiclusterdeploy/sfservicebindingreplicator/sfservicebindingreplicator_controller.go

```diff
@@ -0,0 +1,360 @@
+			}
+		} else if replica.GetState() == "in_queue" || replica.GetState() == "delete" {
+			log.Info("replica in in_queue or delete state, not replicating it to master cluster",
+				"clusterID", clusterID, "bindingID", bindingID, "state", state)
+			return reconcile.Result{}, nil
+		} else {
+			replica.Status.DeepCopyInto(&binding.Status)
+			binding.SetLabels(replica.GetLabels())
+			binding.SetAnnotations(replica.GetAnnotations())
+			replicaLabels = replica.GetLabels()
+			replicaState = replica.GetState()
+		}
+		replicaLastOperation, ok := replicaLabels[constants.LastOperationKey]
+		if !ok {
+			replicaLastOperation = "in_queue"
+		}
+		//Replicate the binding secret from right to left if state is succeeded
+		if replicaState == "succeeded" {
+			if replicaLastOperation == "delete" {
+				//delete the secret
+				log.Info("unbind on sister cluster completed, deleting secret from master cluster..",
+					"clusterID", clusterID, "bindingID", bindingID, "state", state)
+				secretName := replica.Status.Response.SecretRef
+				if secretName == "" {
+					secretName = "sf-" + binding.GetName()
+				}
+				bindingSecret := &corev1.Secret{}
+				bindingSecret.SetName(secretName)
+				bindingSecret.SetNamespace(binding.GetNamespace())
+				err = r.Delete(context.TODO(), bindingSecret)
+				if err != nil {
+					if apiErrors.IsNotFound(err) {
+						log.Error(err, "Seems like binding secret is already deleted", "bindinID", bindingID)
+					} else {
+						log.Error(err, "Failed to delete secret in master cluster", "binding", bindingID,
+							"clusterID ", clusterID, "state ", state)
+						// Error deleting the object - requeue the request.
+						return reconcile.Result{}, err
+					}
+				}
+			} else {
+				replicaSecret := &corev1.Secret{}
+				secretName := replica.Status.Response.SecretRef
+				if secretName == "" {
+					secretName = "sf-" + binding.GetName()
+				}
+				err = targetClient.Get(context.TODO(), types.NamespacedName{
+					Name:      secretName,
+					Namespace: replica.GetNamespace(),
+				}, replicaSecret)
+				if err != nil {
+					log.Error(err, "Failed to get secret from sister cluster", "binding", bindingID,
+						"clusterID ", clusterID, "state ", state)
+					return reconcile.Result{}, err
+				}
+				bindingSecret := &corev1.Secret{}
+				bindingSecret.Data = make(map[string][]byte)
+				bindingSecret.SetName(replicaSecret.GetName())
+				bindingSecret.SetNamespace(replicaSecret.GetNamespace())
+				for k, v := range replicaSecret.Data {
+					bindingSecret.Data[k] = v
+				}
+				if err = controllerutil.SetControllerReference(binding, bindingSecret, r.scheme); err != nil {
+					log.Error(err, "failed to set owner reference for secret", "binding", bindingID)
+					return reconcile.Result{}, err
+				}
+				log.Info("bind on sister cluster completed, replicating secret to master cluster..",
+					"clusterID", clusterID, "bindingID", bindingID, "state", state)
+				err = r.Update(context.TODO(), bindingSecret)
```

Comment:
Wouldn't this always fail? Even in the case where the secret already existing in the target because of `metadata.generation`?

Shouldn't we do a `Get` first and then decide if we want to make a `Create` or `Update` call?

Reply:
done.

Reply:
For the core resources the client handles it and it wont fail. But for crds or extension resources it fails. For consistency we will first fetch and update. @jayeshjk is working on it